### PR TITLE
qemu.tests: Add negative test case for cluster_size_check

### PR DESCRIPTION
**Pull request by Chenditang**
1. test some invalid cluster_size
2. fix AttributeError: 'QemuImg' object has no attribute 'params'

### CODE REVIEW

**Comment by Chenditang** [comment]
The test result on libvirt-1.2.13 fc20.x86_64

\# ./run -k --no-downloads -t qemu --tests cluster_size_check
SETUP: PASS (0.40 s)
DATA DIR: /var/lib/virt_test
DEBUG LOG: /root/autotest/client/tests/virt/logs/run-2015-08-24-13.02.13/debug.log
TESTS: 2
(1/2) type_specific.io-github-autotest-qemu.cluster_size_check.positive_testing: PASS (0.79 s)
(2/2) type_specific.io-github-autotest-qemu.cluster_size_check.negative_testing: FAIL (0.62 s)
TOTAL TIME: 1.67 s
TESTS PASSED: 1
TESTS FAILED: 1
SUCCESS RATE: 50.00 %

**Comment by Chenditang** [comment]
Create image with "cluster_size=0" should return failure:
\# qemu-img create -f qcow2 -o cluster_size=0 test.img 1G
Formatting 'test.img', fmt=qcow2 size=1073741824 encryption=off cluster_size=0 lazy_refcounts=off 
\# echo $?
0
\# qemu-img info test.img 
image: test.img
file format: qcow2
virtual size: 1.0G (1073741824 bytes)
disk size: 196K
cluster_size: 65536

**Comment by vipmike007** [comment]
Hi, I failed to review this patch for the negtive testing
08/25 15:56:00 DEBUG|qemu_stora:0450| Checking image file /root/autotest/test/autotest-devel/client/tests/virt/shared/data/images/cluster_size_check.qcow2
08/25 15:56:00 DEBUG|     utils:0114| Running '/bin/qemu-img info /root/autotest/test/autotest-devel/client/tests/virt/shared/data/images/cluster_size_check.qcow2'
08/25 15:56:00 DEBUG|     utils:0114| Running '/bin/qemu-img check /root/autotest/test/autotest-devel/client/tests/virt/shared/data/images/cluster_size_check.qcow2'
08/25 15:56:00 INFO |env_proces:0315| Remove image on local fs.
08/25 15:56:00 DEBUG|qemu_stora:0357| Removing image file /root/autotest/test/autotest-devel/client/tests/virt/shared/data/images/cluster_size_check.qcow2
08/25 15:56:00 DEBUG|env_proces:1283| Thread quit. Used to failed to get register info from guest 20150825-152800-NbCMtqbP for 2 times.
08/25 15:56:00 DEBUG|     utils:0114| Running 'true'
08/25 15:56:00 DEBUG|     utils:0114| Running 'ps -o comm 1'
08/25 15:56:00 DEBUG|     utils:0114| Running 'lsmod |grep ksm'
08/25 15:56:00 DEBUG|     utils:0114| Running 'ps -C ksmtuned -o pid='
08/25 15:56:00 DEBUG|     utils:0114| Running 'cat /sys/kernel/mm/ksm/run'
08/25 15:56:00 DEBUG|     utils:0114| Running 'cat /sys/kernel/mm/ksm/pages_to_scan'
08/25 15:56:00 DEBUG|     utils:0114| Running 'cat /sys/kernel/mm/ksm/sleep_millisecs'
08/25 15:56:00 DEBUG|     utils:0114| Running 'echo 5000 > /sys/kernel/mm/ksm/pages_to_scan'
08/25 15:56:00 DEBUG|     utils:0114| Running 'echo 1 > /sys/kernel/mm/ksm/run'
08/25 15:56:00 DEBUG|     utils:0114| Running 'echo 50 > /sys/kernel/mm/ksm/sleep_millisecs'
08/25 15:56:00 DEBUG|     utils:0114| Running 'dmesg'
08/25 15:56:00 ERROR|      test:0414| Exception escaping from test:
Traceback (most recent call last):
  File "/root/autotest/test/autotest-devel/client/shared/test.py", line 411, in _exec
    _call_test_function(self.execute, *p_args, **p_dargs)
  File "/root/autotest/test/autotest-devel/client/shared/test.py", line 823, in _call_test_function
    return func(*args, **dargs)
  File "/root/autotest/test/autotest-devel/client/shared/test.py", line 291, in execute
    postprocess_profiled_run, args, dargs)
  File "/root/autotest/test/autotest-devel/client/shared/test.py", line 209, in _call_run_once
    *args, **dargs)
  File "/root/autotest/test/autotest-devel/client/shared/test.py", line 313, in run_once_profiling
    self.run_once(*args, **dargs)
  File "/root/autotest/test/autotest-devel/client/tests/virt/virt.py", line 155, in run_once
    run_func(self, params, env)
  File "/root/autotest/test/autotest-devel/client/shared/error.py", line 141, in new_fn
    return fn(*args, **kwargs)
  File "/root/autotest/test/autotest-devel/client/tests/virt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/cluster_size_check.py", line 107, in run
    % (fail, fail_log))
TestFail: Cluster size check failed 1 times:
Run successfully with wrong options!
    [context: Finall result check]

**Comment by vipmike007** [comment]
For the negtive testing ,I think we need to capture the fail log and analyze it.
if the test should go to pass at last if it meeting the fail criteria.

**Comment by xutian** [comment]
@Chenditang 

Any update  for @vipmike007 's comment ?

thanks,
Xu

**Comment by Chenditang** [comment]
@xutian @vipmike007 
Updated, Thanks~~

**Comment by xutian** [comment]
@Chenditang   view from your code, duplicated error log info $fail_log will print when test failed.

File: qemu/tests/cluster_size_check.py [hunk 1]
```diff
@@ -45,29 +44,38 @@ def check_cluster_size(parttern, expect, csize_set):
...
-                fail_log += "Cluster size mismatch when set it to "
-                fail_log += "%s.\n" % csize_set
         else:
-            logging.error("Can not get the cluster size from command: %s"
-                          % output)
-            cfail += 1
-            fail_log += "Can not get the cluster size from command:"
-            fail_log += " %s\n" % output
+            if result.exit_status == 1:
+                logging.error("Create image with cluster size return failure!")
```
Comment (xutian) [inline qemu/tests/cluster_size_check.py]: Noise...

File: qemu/tests/cluster_size_check.py [hunk 2]
```diff
@@ -45,29 +44,38 @@ def check_cluster_size(parttern, expect, csize_set):
...
+                logging.error("%s.\n" % result)
```
Comment (xutian) [inline qemu/tests/cluster_size_check.py]: same with above

File: qemu/tests/cluster_size_check.py [hunk 3]
```diff
@@ -45,29 +44,38 @@ def check_cluster_size(parttern, expect, csize_set):
...
+                cfail += 1
+            else:
+                output = image.info()
+                error.context("Check the cluster size from output", logging.info)
+                cluster_size = re.findall(parttern, output)
+                if cluster_size:
+                    if cluster_size[0] != expect:
+                        logging.error("Cluster size mismatch when set it to %s."
```
Comment (xutian) [inline qemu/tests/cluster_size_check.py]: better to print what's the expect what's the values of current.

**Comment by Chenditang** [comment]
@xutian 
New version, Thanks~~

**Comment by xutian** [comment]
looks good to me. merged . thanks @Chenditang

**Pull request merged**